userspace/libscap/engine/kmod/scap_kmod.c

```diff
@@ -247,8 +248,8 @@ int32_t scap_kmod_init(scap_t *handle, scap_open_args *oargs)
 		return SCAP_FAILURE;
 	}
 
-	handle->m_ncpus = sysconf(_SC_NPROCESSORS_CONF);
```

Comment:
could we remove the `handle->m_ncpus` field from the `scap` struct

```c
struct scap
{
	const struct scap_vtable *m_vtable;
	struct scap_engine_handle m_engine;

	scap_mode_t m_mode;
	char m_lasterr[SCAP_LASTERR_SIZE];

	// Used for scap_strerror
	char m_strerror_buf[SCAP_LASTERR_SIZE];

	struct scap_proclist m_proclist;
	scap_mountinfo* m_dev_list;
	scap_threadinfo m_fake_kernel_proc;
	uint64_t m_evtcnt;
	scap_addrlist* m_addrlist;
	scap_machine_info m_machine_info;
	scap_userlist* m_userlist;
	struct ppm_proclist_info* m_driver_procinfo;
	bool refresh_proc_table_when_saving;
	uint32_t m_fd_lookup_limit;
	uint32_t m_ncpus;
```

Reply:
🤦 Thanks for spotting this!